### PR TITLE
Adding Result

### LINKED ISSUE #366: Result extensions
### Description

It would be nice to have Error handling with Result. Perhaps even c# extensions.

### Expected behavior

```c#
            var a = FSharpResult<int, string>.NewOk(1);
            var c = a.Match(i => 0, _ => 1);
            Assert.AreEqual(0, c);
```

### PR DESCRIPTION
**Pull request by wallymathieu**
This fixes #366 
Part of the error handling code for `Choice` ported to `Result`

### CODE REVIEW

**Comment by wallymathieu** [comment]
@panesofglass does this makes sense? What is your thoughts?

File: src/FSharpx.Extras/ComputationExpressions/Result.fs [hunk 1]
```diff
@@ -0,0 +1,76 @@
+namespace FSharpx
+
+module Result =
```
Comment (panesofglass) [inline src/FSharpx.Extras/ComputationExpressions/Result.fs]: This is missing the actual computation expression builder.

File: tests/FSharpx.CSharpTests/ResultTests.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
+using System;
+using Microsoft.FSharp.Core;
+using NUnit.Framework;
+
+namespace FSharpx.CSharpTests {
```
Comment (panesofglass) [inline tests/FSharpx.CSharpTests/ResultTests.cs]: It does not appear that this file was added to the test project. Also, might you consider adding tests for the CE and operators?

**Comment by panesofglass** [comment]
Thank you for the contribution, @wallymathieu! I left a few comments that we should address before merging.

**Comment by wallymathieu** [comment]
Thanks for the review! 😀

File: tests/FSharpx.CSharpTests/ResultTests.cs (hunk 2 above)
Comment (wallymathieu) [reply tests/FSharpx.CSharpTests/ResultTests.cs]: New style csproj projects don't require file includes.

File: src/FSharpx.Extras/ComputationExpressions/Result.fs (hunk 1 above)
Comment (wallymathieu) [reply src/FSharpx.Extras/ComputationExpressions/Result.fs]: I've added a CE builder

**Pull request merged**